python-threatexchange/threatexchange/cli/cli_state.py

```diff
@@ -0,0 +1,198 @@
+from threatexchange.cli.exceptions import CommandError
+from threatexchange.fetcher.collab_config import CollaborationConfigBase
+from threatexchange.fetcher.fetch_state import (
+    FetchCheckpointBase,
+    FetchedSignalMetadata,
+)
+from threatexchange.fetcher.simple import state as simple_state
+from threatexchange.fetcher.fetch_api import SignalExchangeAPI
+from threatexchange.signal_type import signal_base
+from threatexchange.signal_type import index
+
+
+class FetchCheckpoint(t.NamedTuple):
+    last_full_fetch: float
+    last_fetch: float
+
+
+class Dataset:
+
+    EXTENSION = ".te"
+
+
+class CliIndexStore:
+
+    FILE_EXTENSION = ".index"
+
+    def __init__(self, indice_dir: pathlib.Path) -> None:
+        self.dir = indice_dir
+
+    def get_available(self) -> t.List[str]:
+        return [
+            str(f)[-len(self.FILE_EXTENSION)]
+            for f in self.dir.glob(f"*{self.FILE_EXTENSION}")
+        ]
+
+    def clear(
+        self, only_types: t.Optional[t.Iterable[t.Type[SignalType]]] = None
+    ) -> None:
+        only_names = None
+        if only_types is not None:
+            only_names = {st.get_name() for st in only_types}
+        for file in self.dir.glob(f"*{self.FILE_EXTENSION}"):
+            if (
+                only_names is None
+                or str(file)[: -len(self.FILE_EXTENSION)] in only_names
+            ):
+                logging.info("Removing index %s", file)
+                file.unlink()
+
+    def _index_file(self, signal_type: t.Type[signal_base.SignalType]) -> pathlib.Path:
+        return self.dir / f"{signal_type.get_name()}{self.FILE_EXTENSION}"
+
+    def store_index(
+        self, signal_type: t.Type[signal_base.SignalType], index: SignalTypeIndex
+    ) -> None:
+        assert signal_type.get_index_cls() == index.__class__
+        path = self._index_file(signal_type)
+        with path.open("wb") as fout:
+            index.serialize(fout)
+
+    def load_index(
+        self, signal_type: t.Type[signal_base.SignalType]
+    ) -> t.Optional[index.SignalTypeIndex]:
+        path = self._index_file(signal_type)
+        if not path.exists():
+            return None
+        with path.open("rb") as fin:
+            return signal_type.get_index_cls().deserialize(fin)
+
+
+class CliSimpleState(simple_state.SimpleFetchedStateStore):
+
+    JSON_CHECKPOINT_KEY = "checkpoint"
+    JSON_RECORDS_KEY = "records"
+
+    def __init__(
+        self, api_cls: t.Type[SignalExchangeAPI], fetched_state_dir: pathlib.Path
+    ) -> None:
+        super().__init__(api_cls)
+        self.dir = fetched_state_dir
+
+    def collab_file(self, collab_name: str) -> pathlib.Path:
+        return self.dir / f"{collab_name}.state.json"
+
+    def clear(self, collab: CollaborationConfigBase) -> None:
+        file = self.collab_file(collab.name)
+        if file.is_file():
+            logging.info("Removing %s", file)
+            file.unlink(missing_ok=True)
+        if file.parent.is_dir():
```

Comment:
When can file.parent.is_dir() be False?

Reply:
Happened in testing where parent folder was missing